nixos/modules/security/apparmor.nix

```
@@ -42,22 +51,55 @@ in
         description = ''
           AppArmor policies.
         '';
-        type = types.attrsOf (types.submodule ({ name, config, ... }: {
-          options = {
-            enable = mkDisableOption "loading of the profile into the kernel";
-            enforce = mkDisableOption "enforcing of the policy or only complain in the logs";
-            profile = lib.mkOption {
-              description = "The policy of the profile.";
-              type = types.lines;
-              apply = pkgs.writeText name;
-            };
-          };
-        }));
-        default = {};
+        type = types.attrsOf (
+          types.submodule (
+            {
+              name,
+              config,
+              options,
+              ...
+            }:
+            {
+              options = {
+                state = lib.mkOption {
+                  description = "The state of the profile as applied to the system by nix";
+                  type = types.enum [
+                    "disable"
+                    "complain"
+                    "enforce"
+                  ];
+                  # should enforce really be the default?
+                  # the docs state that this should only be used once one is REALLY sure nothing's gonna break
+                  default = "enforce";
+                };
+
+                profile = lib.mkOption {
+                  description = "The policy of the profile. Incompatible with path.";
```

```suggestion
                  description = "The policy of the profile, i.e. the desired content of `/etc/apparmor.d/<name>.conf`. Incompatible with path.";
```

I'm not very familiar with apparmor but the current formulation is really obscure to me.